File: Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace aliment_backend.Entities
{
    /// <summary>
    /// Représente un produit.
    /// </summary>
    public class Product
    {
        /// <summary>
        /// Identifiant unique du produit.
        /// </summary>
        [Key]
        public int Id { get; set; }

        /// <summary>
        /// Nom du produit.
        /// </summary>
        [Required(ErrorMessage = "Le nom est requis.")]
        [StringLength(30, ErrorMessage = "Le nom doit comporter entre {2} et {1} caractères.", MinimumLength = 3)]
        public string? Name { get; set; }

        /// <summary>
        /// Description du produit.
        /// </summary>
        [MaxLength(250)]
        public string? Description { get; set; }

        /// <summary>
        /// Durée de conservation du produit.
        /// </summary>
        [Required(ErrorMessage = "La durée de conservation est requise.")]
        [Range(1, 25, ErrorMessage = "La durée de conservation doit être comprise entre {1} et {2}.")]
        public int DurationConservation { get; set; }

        /// <summary>
        /// Clé étrangère faisant référence à la catégorie à laquelle appartient ce produit.
        /// </summary>
        [ForeignKey("Category")]
        public int CategoryId { get; set; }

        /// <summary>
        /// Collection des stocks associés à ce produit.
        /// </summary>
        public ICollection<Stock> Stocks { get; set; } = new List<Stock>();

        /// <summary>
        /// Catégorie à laquelle appartient ce produit.
        /// </summary>
        public Category? Category { get; set; }

        /// <summary>
        /// Nombre de produits en stock pour ce produit.
        /// </summary>
        [NotMapped]
        private int _nbProductStock;

        /// <summary>
        /// Propriété calculée représentant le nombre de produits en stock pour ce produit.
        /// </summary>
        [NotMapped]
        public int NbProductStock
        {
            get => Stocks.Count;
            set => _nbProductStock = value;
        }
    }
}

File: src/app/features/admin/child/_model/product.ts
import { Category } from "./category";

/**
 * La classe `Product` représente un produit alimentaire dans l'application.
 * Elle stocke des informations sur un produit, telles que son identifiant, son nom, sa description,
 * la durée de conservation, la catégorie à laquelle il appartient, l'identifiant de la catégorie,
 * et le nombre de produits en stock.
 */
export class Product {
  /**
   * Identifiant unique du produit.
   * @type {number}
   */
  id!: number;

  /**
   * Nom du produit.
   * @type {string}
   */
  name!: string;

  /**
   * Description du produit.
   * @type {string}
   */
  description!: string;

  /**
   * Durée de conservation du produit en jours.
   * @type {number}
   */
  durationConservation!: number;

  /**
   * Objet représentant la catégorie à laquelle le produit appartient.
   * @type {Category}
   */
  category!: Category;

  /**
   * Identifiant unique de la catégorie à laquelle le produit appartient.
   * @type {number}
   */
  categoryId!: number;

  /**
   * Nombre de produits en stock pour ce produit.
   * @type {number}
   */
  nbProductStock!: number;
}
